### PR TITLE
feat [IDP-1510] : Added the tutorial to add links in software components

### PR DESCRIPTION
**Pull request by Debanitrkl**
# Harness Developer Pull Request
Added the tutorial to add links in software components

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: tutorials/internal-developer-portal/add-links-in-components.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+---
+title: How to add Links in Software Components
+description: Steps to add various links associated with the component.
+sidebar_position: 4
+---
+
+## Introduction
+In this tutorial we will explore on how to add links associated with the Software Components which includes user profiles, groups, dashboards, support pages, and more. Each link can be accompanied by an icon for visual representation.
```
Comment (OrkoHunter) [inline tutorials/internal-developer-portal/add-links-in-components.md:8]:
```suggestion
In this tutorial we will explore how to add links associated with the Software Components which could be dashboards, support pages, runbooks, email groups, Slack channels, etc. Each link is an arbitrary name and URL pair, which can be accompanied by an icon for visual representation.
```

File: tutorials/internal-developer-portal/add-links-in-components.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+Once you register a Software Component in the Catalog this is a sample of how your overview section would look based on your IDP yaml. 
+
+![](./static/link-card.png)
+
+In the depicted image, you'll notice a **Links Card** that serves as a navigational tool. These links can be anything from the following suggested categories:
```
Comment (OrkoHunter) [inline tutorials/internal-developer-portal/add-links-in-components.md:14]:
```suggestion
In the depicted image, you'll notice a **Links Card** that serves as a navigational tool. These links can be anything. Here are some examples -
```

File: tutorials/internal-developer-portal/add-links-in-components.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ 2. Now add **links** to the IDP yaml as shown in the example below.
+    
+<details>
+<summary>Link Attributes</summary>
+
+- **url:** This is the actual web address or URL that the link points to. It's the destination users will be directed to when they click on the link.
+
+- **title:** This serves as the display name or label for the link. It gives users a brief idea of what to expect when they click on the link or where it might lead them.
+
+- **icon:** An optional attribute, you can select from a category of supported icon that provides a visual representation or hint about the nature or category of the link.
```
Comment (OrkoHunter) [inline tutorials/internal-developer-portal/add-links-in-components.md:42]: Is there a place where we can see all available links? I suppose this is MUI icons. cc @jenil-harness

**Review by OrkoHunter — APPROVED** [review]

File: tutorials/internal-developer-portal/add-links-in-components.md (hunk 3 above)
Comment (jenil-harness) [reply tutorials/internal-developer-portal/add-links-in-components.md:42]: We are going to replace this with harness icons. I will share more details once the implementation is done

**Pull request merged**